target/linux/ipq40xx/image/generic.mk

```diff
@@ -780,6 +780,25 @@ define Device/qxwlan_e2600ac-c2
 endef
 TARGET_DEVICES += qxwlan_e2600ac-c2
 
+define Device/teltonika_rutx10
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Teltonika
+	DEVICE_MODEL := RUTX10
+	SOC := qcom-ipq4018
+	DEVICE_DTS_CONFIG := config@5
+	KERNEL_INSTALL := 1
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	FILESYSTEMS := squashfs
+	SUPPORTED_DEVICES := teltonika,rutx10 teltonika,rutx
```

Comment:
First string is default, what's the second needed for?
Probably this line can be removed completely ...

Reply:
This is needed for the stock firmware to accept the factory image. Stock firmware checks for `teltonika,rutx` in metadata.

Reply:
If possible, this should be handled by a factory image instead.

Reply:
This is handled by the factory image now